sql/core/src/test/scala/org/apache/spark/sql/connector/MergeIntoTableSuiteBase.scala

```diff
@@ -32,6 +32,58 @@ abstract class MergeIntoTableSuiteBase extends RowLevelOperationSuiteBase {
 
   import testImplicits._
 
+  test("merge into table containing added column with default value") {
+    withTempView("source") {
+      sql(
+        s"""CREATE TABLE $tableNameAsString (
+           | pk INT NOT NULL,
+           | salary INT NOT NULL DEFAULT -1,
+           | dep STRING)
+           |PARTITIONED BY (dep)
+           |""".stripMargin)
+
+      append("pk INT NOT NULL, dep STRING",
+        """{ "pk": 1, "dep": "hr" }
+          |{ "pk": 2, "dep": "hr" }
+          |{ "pk": 3, "dep": "hr" }
+          |""".stripMargin)
+
+      sql(s"ALTER TABLE $tableNameAsString ADD COLUMN txt STRING DEFAULT 'initial-text'")
+
+      checkAnswer(
+        sql(s"SELECT * FROM $tableNameAsString"),
+        Seq(
+          Row(1, -1, "hr", "initial-text"),
+          Row(2, -1, "hr", "initial-text"),
+          Row(3, -1, "hr", "initial-text")))
+
+      val sourceRows = Seq(
+        (1, 100, "hr"),
+        (4, 400, "hr"))
+      sourceRows.toDF("pk", "salary", "dep").createOrReplaceTempView("source")
+
+      sql(
+        s"""MERGE INTO $tableNameAsString t
+           |USING source s
+           |ON t.pk = s.pk
+           |WHEN MATCHED THEN
+           | UPDATE SET t.salary = s.salary, t.txt = DEFAULT
```

Comment:
Can we remove `t.txt = DEFAULT` part in this line? Or, can we have a different value instead of `DEFAULT`?

Reply:
I actually added it intentionally to verify `DEFAULT` are properly resolved in assignments. Let me know if this makes sense.

Reply:
Got it.

Reply:
Could you add another `MERGE INTO` without `t.txt = DEFAULT` to have that test coverage? If we already have, could you give me the link to the test case?

Reply:
I believe the line below should already cover it (we omit DEFAULT for `txt` but not for `salary`):
```
WHEN NOT MATCHED THEN
 INSERT (pk, salary, dep) VALUES (s.pk, DEFAULT, s.dep)
```

Essentially, this test is structured so that the table initially has a column with a default value and we add one more column with a default value later. The MERGE already verifies explicit DEFAULT and missing columns.

Reply:
Let me know if I understood your question correctly. Happy to add a new test if needed.

Reply:
@dongjoon-hyun, gentle ping to see if my answer makes sense.